### ISSUE TITLE
no-control-regex doesn't catch erros on escaped strings

### ISSUE THREAD

**Issue by IllusionMH** [microsoft/tslint-microsoft-contrib#785]
### Bug Report

-   **`tslint-microsoft-contrib` version**: 6.0.0
-   **TSLint version**: 5.12.0
-   **TypeScript version**: 3.1.1
-   **Running TSLint via**: CLI

Difference in behavior during test mighration mentioned in [comment](https://github.com/Microsoft/tslint-microsoft-contrib/pull/665#discussion_r246965219)

#### TypeScript code being linted
```tsx
// should fail
new RegExp('something \x1f something else'); // creates regexp with special character in it (equivalent of /something  something else/)

new RegExp('something \\x1f something else'); // creates regexp with character match sequence in it (equivalent of /something \x1f something else/)

new RegExp('something \\\x1f something else'); // creates regexp that contains backslash followed by special character in it (equivalent of /something \ something else/)

// should pass
new RegExp('something \\\\x1f something else'); // creates regexp that correctly escaped backslash followed by characters x,1,f (equivalent of /something \\x1f something else/)
```
with `tslint.json` configuration:

```json
{
    "rules": {
        "no-control-regex": true
    }
}
```

#### Actual behavior

Detected errors in strings with 1 and 3 `\` characters (will also fail on 5 characters)
No errors in strings with 2 and 4 `\` characters

#### Expected behavior

Should detect errors in strings with 1, 2, and 3 `\` characters
No error in string with  4 `\` characters

_ESLint behavior (was prototype):_

Will detect errors in strings with 1, 2, and 5 `\` characters
Will detect no errors in strings with 3 and 4 `\` characters.

Not sure why it passes on 3 `\` characters, but it is up for discussion.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #876. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!
